src/nvim/terminal.c

```diff
@@ -460,12 +462,9 @@ static int terminal_execute(VimState *state, int key)
       break;
 
     case K_EVENT:
-      // We cannot let an event free the terminal yet. It is still needed.
-      s->term->refcount++;
       multiqueue_process_events(main_loop.events);
-      s->term->refcount--;
-      if (s->term->buf_handle == 0) {
-        s->close = true;
+      if (s->term->exited) {
```

Comment:
Would this check also be necessary before calling `send_mouse_event()`? In this function is a call to `invalidate_terminal()` that could put the terminal back into the map `invalidated_terminals`.